benchmarks/modules/utils.py

```diff
@@ -267,11 +271,38 @@ def setup_spack_environment(self):
             f'(cd {cp_dir}; find . \( -name "spack.yaml" -o -name "compilers.yaml" -o -name "packages.yaml" \) -print0 | xargs -0 tar cf - | tar -C {dest} -xvf -)',
             f'spack -e {self.build_system.environment} config add "config:install_tree:root:{env_dir}/opt"',
         ]
-
+        # else f"{d[key].value}" if isinstance(d[key].value, bool) for boolean with "True" instead of True
+        cmd_spack_variants = 'from spack import environment; d = environment.active_environment().spec_lists["specs"].specs[0].variants.dict; keys = list(d.keys());values = {key: ([d[key].value[i] for i in range(len(d[key].value))] if isinstance(d[key].value, tuple) else d[key].value) for key in keys};print(values)'
```

Comment:
I think I found it from [documentation](https://spack.readthedocs.io/en/latest/spack.environment.html#spack.environment.Environment.concretized_user_specs): 
```
cmd_spack_disp = 'from spack import environment; d = environment.active_environment().concretized_user_specs;print(d);'
self.postrun_cmds.append(f'echo "DISPLAY: $(spack -e {self.build_system.environment} python -c \'{cmd_spack_disp}\')"')

prints out : 

DISPLAY: [babelstream%gcc@9.2.0+cuda cuda_arch=70,72]
```
I could run an iteration ( instead of just picking 0) and create dictionaries for each _concretized_ spec ?
 @ilectra I think this would also address the issue you mentioned in your comment [here](https://github.com/ukri-excalibur/excalibur-tests/pull/262#discussion_r1512745801) ?

Reply:
Yeah, the loop you're describing sounds right. I've no idea of the thing Mose is talking about.

Remember to adjust the postprocessing accordingly when you add the loop!

Reply:
https://github.com/ukri-excalibur/excalibur-tests/pull/262/commits/f696cd55a0fb2205e4e1923c197061bb134dc09d
@giordano @ilectra 
Input : ( `intel-tbb` and `cuda` are unrelated but I couldn't get `hpl` working on Isambard so I just wanted to test multiple packages in one spec) 
```
$ reframe -c benchmarks/apps/babelstream -r --tag cuda --system=isambard-macs:volta --setvar=num_cpus_per_task=40 -S build_locally=false -Sspack_spec='babelstream%gcc@9.2.0 +cuda cuda_arch=70,72 intel-tbb%gcc@9.2.0'
```

Output : 
```
spack_spec_dict: {'babelstream': {'compiler': {'name': 'gcc', 'version': [9.2.0]}, 'variants': {'cuda': True, 'cuda_arch': ('70', '72')}}, 'intel-tbb': {'compiler': {'name': 'gcc', 'version': [9.2.0]}, 'variants': {}}, 'mpi': ()}
```
so the dictionary is now more easy to access since we would have ` ["babelstream"]["compiler"]["name"] `
Also, if mpi is not present in all list of specs, it will be an empty tuple. I kept `mpi` out in the first level dictionary since it is something extra so anyone who needs to acess `mpi` data just finds it via ["mpi"][i] where i will be the package order

Reply:
#TODO : (for me ) 
- babelstream_compiler_name 
- babelstream_compiler_version
- babelstream_variant
- babelstream_mpi 

4 columns

From perflog_handler.py:
https://github.com/ukri-excalibur/excalibur-tests/blob/c2caf65ae61a08b719b2300e90e62e271593ce46/post-processing/perflog_handler.py#L118C1-L125C43

Add spack_spec_dict to here and then use another loop to parse one level down ( do not parse variants since we don't know for each app )